index.js

```diff
@@ -35,31 +55,85 @@ module.exports = {
 
   _debugTree() {
     if (!this._cachedDebugTree) {
-      this._cachedDebugTree = require('broccoli-debug').buildDebugCallback(`ember-cli-babel:${this._parentName()}`);
+      this._cachedDebugTree = require('broccoli-debug').buildDebugCallback(`ember-cli-babel:${_parentName(this.parent)}`);
     }
 
     return this._cachedDebugTree.apply(null, arguments);
   },
 
+  /**
+   * Default babel options
+   * @param {*} config 
+   */
+  _getDefaultBabelOptions(config = {}) {
+     let emberCLIBabelConfig = config["ember-cli-babel"];
+     let providedAnnotation;
+     let throwUnlessParallelizable;
+     let sourceMaps = false;
+     let shouldCompileModules = _shouldCompileModules(config, this.project);
+
+     if (emberCLIBabelConfig) {
+       providedAnnotation = emberCLIBabelConfig.annotation;
+       throwUnlessParallelizable = emberCLIBabelConfig.throwUnlessParallelizable;
+     }
+ 
+     if (config.babel && "sourceMaps" in config.babel) {
+       sourceMaps = config.babel.sourceMaps;
+     }
+
+     let options = {
+       annotation: providedAnnotation || `Babel: ${_parentName(this.parent)}`,
+       sourceMaps,
+       throwUnlessParallelizable,
+       filterExtensions: _getExtensions(config, this.parent),
+       plugins: []
+     };
+
+     if (shouldCompileModules) {
+       options.moduleIds = true;
+       options.getModuleId = require("./lib/relative-module-paths").getRelativeModulePath;
+     }
+
+     options.highlightCode = _shouldHighlightCode(this.parent);
+     options.babelrc = false;
+
+     return options;
+  },
+
   transpileTree(inputTree, _config) {
+
     let config = _config || this._getAddonOptions();
     let description = `000${++count}`.slice(-3);
     let postDebugTree = this._debugTree(inputTree, `${description}:input`);
-
-    let options = this.buildBabelOptions(config);
+    let options = this._getDefaultBabelOptions(config);
     let output;
-    if (this._shouldDoNothing(options)) {
+
+    const babelConfigFile = ROOT_CONFIG_FILENAMES.find((fileName) =>
+      fs.existsSync(path.resolve(this.parent.root, fileName))
+    );
+
+    const customAddonConfig = config['ember-cli-babel'];
+    const shouldUseBabelConfigFile = customAddonConfig && customAddonConfig.useBabelConfig && babelConfigFile;
+
+    if (shouldUseBabelConfigFile) {
+      options = Object.assign({}, options, {
+        "presets":[ require.resolve(path.resolve(this.parent.root, babelConfigFile)) ],
```

Comment:
Isn't babelConfigFile here already resolved?

Reply:
no, the `babelConfigFile` is only the file name that was found based on the path. I ll change that part so that it returns the whole path instead of just the file name.